File: application/src/main/java/project/shipment/ShipmentRepository.java
package project.shipment;

import java.util.ArrayList;
import java.util.List;

public class ShipmentRepository {
    private static List<Shipment> repository=new ArrayList<Shipment>();
    private static int nextIdShipment;

    public ShipmentRepository() {
        repository = new ArrayList<>();
        nextIdShipment = 0;
    }

    public static int nextIdShipment() {
        return nextIdShipment++;
    }

    public static void saveShipment(Shipment shipment) {
        repository.add(shipment);
    }

    public static void updateShipment(int id, String change, String newchange) {
        if (!repository.contains(repository.get(id))) {
            throw new IllegalAccessError("Shipment not found");
        }
        Shipment shipment = repository.get(id);
        shipment.updateShipment(change, newchange);
    }

    public static Shipment getShipmentById(int id) {
        return repository.get(id);
    }

    public static String getAllShipments() {
        String result="";
        for (Shipment shipment : repository) {
            result += shipment.toString()+"\n";
        }
        return result;
    }

    public static void deleteShipment(int id) {
        if(!repository.contains(repository.get(id))){
            throw new IllegalAccessError("Shipment not found");
        }
        repository.remove(id);
    }
}

File: CPProject/shipmentRepository.cpp
#include "shipmentRepository.hpp"

int ShipmentRepository::nextId_ = 0;

std::list<Shipment> ShipmentRepository::repository = std::list<Shipment>();

int ShipmentRepository::nextIdShipment()
{
      return nextId_++;
}

void ShipmentRepository::addShipment(Shipment shipment)
{
      repository.push_back(shipment);
}

void ShipmentRepository::removeShipment(Shipment shipment)
{
      repository.remove(shipment);
}

void ShipmentRepository::updateShipment(int id, std::string what, std::string change)
{
      for(std::list<Shipment>::iterator it = repository.begin(); it != repository.end(); ++it)
      {
            if (it->getIdShipment() == id)
            {
                  it->updateShipment(what, change);
            }
      }
}

Shipment ShipmentRepository::getShipmentbyId(int id)
{
      for(std::list<Shipment>::iterator it = repository.begin(); it != repository.end(); ++it)
      {
            if (it->getIdShipment() == id)
            {
                  return *it;
            }
      }
      return *repository.end();
}

std::string ShipmentRepository::getAllShipments()
{
      std::string result = "";
      for(std::list<Shipment>::const_iterator it = repository.begin(); it != repository.end(); ++it)
      {
            result += it->display_shipment_info() + "\n";
      }
      return result;
}